src/compas_fab/robots/semantics.py

```diff
@@ -15,6 +15,49 @@ class RobotSemantics(Data):
 
     The semantic model is based on the
     `Semantic Robot Description Format` (`SRDF`_).
+    Typically, the RobotSemantics objects are created from an SRDF file (using :meth:`from_srdf_file`)
+    or loaded by the backend client (using :meth:`compas_fab.backends.PyBulletClient.load_semantics` or
+    :meth:`compas_fab.backends.RosClient.load_robot`).
+
+    Parameters
+    ----------
+    robot_model : :class:`~compas_fab.robots.RobotModel`
+        The robot model.
+    groups : :obj:`dict` of (:obj:`str`, :obj:`dict` of (``links`` : :obj:`list` of :obj:`str`, ``joints`` : :obj:`list` of :obj:`str`)), optional
+        A nested dictionary defining planning groups.
+        The dictionary structure is as follows:
+
+        - Level 1 keys are planning group names : :obj:`str`.
+        - Level 2 contains only two keys:
+
+            -   ``links`` is a :obj:`list` of :obj:`str` containing link names
+            -   ``joints`` is a :obj:`list` of :obj:`str` containing joint names.
+
+    main_group_name : :obj:`str`, optional
+        The name of the main group.
+    passive_joints : :obj:`list` of :obj:`str`, optional
+        A list of passive joint names.
+    end_effectors : :obj:`list` of :obj:`str`, optional
+        A list of end effector link names.
+    disabled_collisions : :obj:`tuple` of (:obj:`str`, :obj:`str`), optional
+        A set of disabled collision pairs.
+        The order is not important, i.e. the pair `('link1', 'link2')` is the same as `('link2', 'link1')`.
+        Only one pair is needed.
+    group_states : dict, optional
+        A nested dictionary defining named states for a particular group, in terms of joint values.
+        This is useful to define states such as "home" or "folded arms" for the robot.
+        The dictionary structure is as follows:
+
+        - Level 1 keys are planning group names : :obj:`str`.
+        - Level 2 keys are group state names : :obj:`str`.
+        - Level 3 keys are joint names and values are joint values : :obj:`str`.
```

Comment:
Maybe add an example instead of the bullet-point description? Or both?

Reply:
It is very involved in explaining this properly. I think these bullet points is more of a note now.
Maybe we can do the example in a separate rst doc in tutorials.